### PR TITLE
[AM][OPS-10527] add kickout page for deregistered customers

### PR DESCRIPTION
**Pull request by almorcrette**

### CODE REVIEW

File: test/pages/Sandbox.sc [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        |If you are outside the UK: +44 2890 538 192
+        |Before you call, make sure you have:
+        |your VAT registration number. This is 9 numbers, for example, 123456789
+        |your bank details
+        """.stripMargin
+  }
+
+  val expectedLines = Expected.mainText.stripSpaces().split("/n")
+
+  implicit class StringOps(s: String) {
```
Comment (AndyHWChung) [inline test/pages/Sandbox.sc:20]: this seems to be a duplication of something else in `CommonPage` - you could stick it in one common place to save duplication.

File: test/pages/Sandbox.sc [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      .replaceAll("[^\\S\\r\\n]+", " ") //replace many consecutive white-spaces (but not new lines) with one space
+      .replaceAll("[\r\n]+", "\n") //replace many consecutive new lines with one new line
+      .split("\n").map(_.trim) //trim each line
+      .filterNot(_ == "") //remove any empty lines
+      .mkString("\n")
+  }
+
+}
+
+println(SandBox.expectedLines.toString)
```
Comment (AndyHWChung) [inline test/pages/Sandbox.sc:37]: do we want this `println`?

File: app/controllers/action/AuthenticatedAction.scala [hunk 3]
```diff
@@ -41,17 +42,18 @@ class AuthenticatedAction @Inject() (
...
-    else {
-      for {
-        customer <- orchestrator.getCustomerData(mtdVrn.vrn)
-      } yield {
-        customer match {
-          case Some(x) => x.isPartiallyMigrated
-          case None    => false
-        }
-      }
+    customer match {
```
[the remark this answers is not in the document]
Comment (almorcrette) [reply app/controllers/action/AuthenticatedAction.scala]: corrected

File: app/controllers/action/AuthenticatedAction.scala [hunk 4]
```diff
@@ -75,29 +77,33 @@ class AuthenticatedAction @Inject() (
...
         case _                    => throw new InsufficientEnrolments
-
       }
 
-      if (Vrn.isMtdEnroled(typedVrn)) {
-        isPartial(typedVrn).flatMap(isPartialResult =>
+      orchestrator.getCustomerData(typedVrn.vrn).flatMap { customer =>
 
-          block(new AuthenticatedRequest(request, enrolments, if (isPartialResult) ClassicVrn(typedVrn.vrn) else typedVrn, isPartialResult))
+        if (isDeregistered(customer)) throw new DeregistrationException
```
[the remark this answers is not in the document]
Comment (almorcrette) [reply app/controllers/action/AuthenticatedAction.scala]: removed new exception, straight to redirect

File: app/model/des/CustomerInformation.scala [hunk 5]
```diff
@@ -105,6 +115,23 @@ object PPOB {
...
 }
 
+final case class Deregistration(
+    deregistrationReason:     Option[String],
+    effectDateOfCancellation: Option[LocalDate],
+    lastReturnDueDate:        Option[LocalDate]
+) {
+
+  val hasDeregistrationReason: Boolean = deregistrationReason match {
+    case Some(reason) if reason != "" => true
```
[the remark this answers is not in the document]
Comment (almorcrette) [reply app/model/des/CustomerInformation.scala]: simplified to `.isEmpty` and inlined

File: test/pages/Sandbox.sc (hunk 1 above)
Comment (almorcrette) [reply test/pages/Sandbox.sc]: removed `Sandbox`, which should not have made final code

File: test/pages/Sandbox.sc (hunk 2 above)
Comment (almorcrette) [reply test/pages/Sandbox.sc]: removed `Sandbox`, which should not have made final code

**Review by AndyHWChung — APPROVED** [review]

**Pull request merged**